src/physics/em/detail/Rayleigh.hh

```diff
@@ -9,30 +9,121 @@
 
 #include "base/Macros.hh"
 #include "base/Types.hh"
+#include "base/Collection.hh"
+#include "physics/base/Types.hh"
 
 namespace celeritas
 {
+struct ModelInteractPointers;
+
 namespace detail
 {
+//---------------------------------------------------------------------------//
+/*!
+ * Rayleigh angular parameters (form factor) for sampling the angular
+ * distribution of coherently scattered photon
+ */
+struct RayleighData
+{
+    static const unsigned int num_parameters = 9;
+    static const unsigned int num_elements   = 100;
+
+    static const real_type angular_parameters[num_parameters][num_elements];
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Rayleigh angular parameters to fit tabulated form factors (\em FF)
+ * \f[
+ *  FF(E,cos)^2 = \Sigma_{j} \frac{a_j}{[1 + b_j x]^{n}}
+ * \f]
+ * where \f$ x= E^{2}(1-cos\theta) \f$ and \em n is the high energy slope of
+ * the form factor and \em a and \em b are free parameters to obtain the best
+ * fit to the form factor. The unit for the energy (\em E) is in MeV.
+ */
+struct RayleighElementData
+{
+    Real3 a;
+    Real3 b;
+    Real3 n;
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Storage for Rayleigh angular parameters
+ */
+template<Ownership W, MemSpace M>
+struct RayleighParameters
+{
+    using ElementId = celeritas::ItemId<unsigned int>;
+
+    template<class T>
+    using ElementItems = celeritas::Collection<T, W, M, ElementId>;
+
+    ElementItems<RayleighElementData> data;
+
+    explicit CELER_FUNCTION operator bool() const { return !data.empty(); }
+
+    //! Assign from another set of parameters
+    template<Ownership W2, MemSpace M2>
+    RayleighParameters& operator=(const RayleighParameters<W2, M2>& other)
+    {
+        CELER_EXPECT(other);
+        data = other.data;
+
+        return *this;
+    }
+};
+
 //---------------------------------------------------------------------------//
 /*!
  * Device data for creating an interactor.
  */
-struct RayleighInteractorPointers
+template<Ownership W, MemSpace M>
+struct RayleighGroup
 {
-    //! ID of an electron
-    ParticleId electron_id;
+    //! Model ID
+    ModelId model_id;
+
     //! ID of a gamma
     ParticleId gamma_id;
-    // XXX additional data
+
+    //! Rayleigh angular parameters
+    RayleighParameters<W, M> params;
```

Comment:
Sorry that I did not catch that.  I removed the RayleighParameters struct and use RayleighParameters instead of RayleighElementData (just for naming).